### PR TITLE
add extra cases to SizeEqualsZero java hints

### PR DESCRIPTION
**Pull request by bradh**
Adds support for size() > 0, size() < 1, and size() <=0.

I think it would be useful to have unit tests for this, but I need a hint on how to write them.

### CODE REVIEW

File: java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java [hunk 1]
```diff
@@ -51,11 +51,22 @@
...
         @TriggerPattern(value="$subj.size() == 0", 
             constraints = @ConstraintVariableType(type = "java.util.Map", variable = "$subj")),
+        @TriggerPattern(value="$subj.size() <= 0",
+            constraints = @ConstraintVariableType(type = "java.util.Collection", variable = "$subj")),
+        @TriggerPattern(value="$subj.size() <= 0",
+            constraints = @ConstraintVariableType(type = "java.util.Map", variable = "$subj")),
+        @TriggerPattern(value="$subj.size() < 1",
+            constraints = @ConstraintVariableType(type = "java.util.Collection", variable = "$subj")),
+        @TriggerPattern(value="$subj.size() < 1",
+            constraints = @ConstraintVariableType(type = "java.util.Map", variable = "$subj")),
```
Comment (sdedic) [inline java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java:69]: Shouldn't be here also the flipped conditions like `1 > $subj.size()` ?

File: java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java [hunk 2]
```diff
@@ -82,6 +93,23 @@ public static ErrorDescription sizeNotEqualsZero(HintContext ctx) {
...
+        @TriggerPattern(value="0 < $subj.size()",
+            constraints = @ConstraintVariableType(type = "java.util.Collection", variable = "$subj")),
+        @TriggerPattern(value="0 < $subj.size()",
+            constraints = @ConstraintVariableType(type = "java.util.Map", variable = "$subj")),
+    })
+    public static ErrorDescription sizeGreaterZero(HintContext ctx) {
+        if (!ctx.getPreferences().getBoolean(CHECK_GREATER_ZERO, CHECK_NOT_EQUALS_DEFAULT)) {
+            return null;
+        }
+        return sizeEqualsZeroHint(ctx, true);
```
Comment (sdedic) [inline java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java:110]: is `not == true` correct in this invocation ?

File: java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java (hunk 1 above)
Comment (bradh) [reply java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java:69]: I will add them.

File: java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java (hunk 2 above)
Comment (bradh) [reply java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java:110]: I believe so. Its essentially the same logic as the previous set of trigger patterns - another way of saying `sizeNotEqualsZero`.

File: java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java (hunk 1 above)
Comment (bradh) [reply java/java.hints/src/org/netbeans/modules/java/hints/perf/SizeEqualsZero.java:69]: Now added.

**Comment by mbien** [comment]
@bradh unfortunately I overlooked this PR when https://github.com/apache/netbeans/pull/3164 got integrated :(

It is lacking the size() < 1 case though. If you still would like to contribute you could rebase to master and add the case + update the test just like the linked PR did.

You could start a new PR or reuse this one, whatever you prefer.

**Pull request closed without merging**